iml-agent/src/action_plugins/stratagem/action_warning.rs

```diff
@@ -81,3 +95,66 @@ pub fn write_records(
 
     Ok(())
 }
+
+// Read mailbox and build a csv of files. return pathname of generated csv
+//
+pub fn read_mailbox(device: &str, mailbox: &str) -> impl Future<Item = String, Error = ()> {
```

Comment:
This should return `ImlAgentError` so the caller can know if something went wrong.